lib/org/chromium/webidl/IDLFragmentExtractor.js

```diff
@@ -21,96 +22,179 @@ foam.CLASS({
       },
     },
     {
+      class: 'Array',
+      of: 'String',
+      documentation: 'Array of IDL Fragments found in the parsed HTML file.',
       name: 'idlFragments',
-      // description: 'all idlFragments found in the parsed HTML file',
+    },
+    {
+      name: 'tags_',
+      documentation: 'Array of tags extracted from file to be processed.',
+      expression: function(file) {
+        var results = this.lexer_.parseString(file.contents).value;
+        if (!results)
+          throw new Error("IDLFragmentExtractor: IDL Parse was not successful.");
+        return results;
+      },
+    },
+    {
+      name: 'lexer_',
+      factory: function() { return this.HTMLLexer.create(); },
+    },
+    {
+      name: 'tagStack_',
+      documentation: 'Stack used to perform HTML Tag matching.',
+      factory: function() {
+        return this.TagStack.create({ isExcluded: this.isExcluded });
+      },
+    },
+    {
+      name: 'tagMatching_',
+      documentation: 'Used to determine if tag matching is currently performed.',
```

Comment:
A longer description explaining that this is a piece of state that changes as a lex stream processes a file. Also, what is the behaviour in the true and the false case? How do we know to "start tag matching again"?